x-pack/legacy/plugins/ml/public/data_frame/components/source_index_preview/source_index_preview.tsx

```diff
@@ -215,12 +217,38 @@ export const SourceIndexPreview: React.SFC<Props> = React.memo(({ cellClick, que
     } as Dictionary<any>;
 
     const field = indexPattern.fields.find(f => f.name === k);
-    const render = (d: string) => {
+
+    const formatField = (d: string) => {
       return field !== undefined && field.type === KBN_FIELD_TYPES.DATE
         ? formatHumanReadableDateTimeSeconds(moment(d).unix() * 1000)
         : d;
     };
 
+    const render = (d: any) => {
+      if (Array.isArray(d) && d.every(item => typeof item === 'string')) {
+        // If the cells data is an array of strings, return as a comma separated list
+        return d.map(item => formatField(item)).join(', ');
```

Comment:
It might be an idea to limit the number of array values to display in a cell. In the anomalies table for example, with influencer values we show a maximum of 5 in the cell, with a `Show more' link to expand the row where all values can be seen.

Reply:
Updated in 0b063a6.